src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -108,7 +115,7 @@ public static IpcEndpointConfig Parse(string config)
 
             string address = "";
             PortType portType = PortType.Connect;
-            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : TransportType.UnixDomainSocket;
+            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : OperatingSystem.IsAndroid() ? TransportType.TcpClient : TransportType.UnixDomainSocket;
```

Comment:
OperatingSystem.IsAndroid() Currently doesn't work in this repo. I am assuming an older version of dotnet/runtime is being used, so if we want to make this work, we may need to bump the runtime version (may lead to a myriad of changes?)

Reply:
Should we also check if this is running on Mono? Right now Android only runs on Mono, but that may not always be the case.

Reply:
Parsing the string is a convenience for humans that would enter these strings on a tool command-line. If we are only implementing this for test purposes and we are already prepared to invoke internal members, can we make the test construct the IpcEndpointConfig object using its constructor and avoid parsing?